vsintegration/src/FSharp.Editor/Options/SettingsPersistence.fs

```diff
@@ -2,83 +2,95 @@
 
 open System
 open System.Collections.Concurrent
-open System.Runtime.InteropServices
 
 open Microsoft.VisualStudio.Settings
+open Microsoft.VisualStudio.Shell
 
 open Newtonsoft.Json
 
-type IPersistSettings =
-    abstract member LoadSettings : unit -> 't
-    abstract member SaveSettings : 't -> unit
+module Settings =
+    type IPersistSettings =
+        abstract member LoadSettings : unit -> 't
```

Comment:
So, instead of creating another type, how about the following:
1) Add new methods to the existing one
2) Call it SettingsStore/ISettingsStore
3) Uniform naming elsewhere: SettingsStore vs PersistSettings (e.g. this file can be also renamed)

I tried this locally quickly, things at least compile and tests seem to be okay, but I might be missing something obvious.

I mean I'm just looking into ways to further simplify the implementation and yeah I am not a big fan of abstract class hierarchies :D

Reply:
3. Agreed.
4. 1. I'm not sure how to do this nicely. I considered matches on `settingsManager` in every method that uses it but there would be two diverging codepaths in a few places. Anyway I tried to touch this as little as possible as it takes a lot of time to build and debug in actual VS instance and it worked before 🙂.